pilot/pkg/model/context.go

```diff
@@ -209,6 +209,8 @@ var DefaultXdsLogDetails XdsLogDetails = XdsLogDetails{}
 // or no response is preferred.
 type XdsResourceGenerator interface {
 	Generate(proxy *Proxy, push *PushContext, w *WatchedResource, updates *PushRequest) (Resources, XdsLogDetails, error)
+
+	GenerateDeltas(proxy *Proxy, push *PushContext, updates *PushRequest) (Resources, Resources, XdsLogDetails, error)
```

Comment:
> Also since most are not implementing it, doesn't that mean only CDS works with delta (meaning no pods can run?)?

I think I'll make it so the rest of the implementations fall back on `Generate`